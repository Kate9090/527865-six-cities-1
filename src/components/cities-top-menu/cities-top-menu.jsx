import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';

import City from '../city/city.jsx';
import {getCities} from "../../reducer/data/selectors";


import withActiveCard from '../../hocs/with-active-card/with-active-card';
const WrappedCity = withActiveCard(City);


const CitiesTopMenu = (props) => {
  const {cities, onCardClick, onCardMouseOut} = props;

  const _renderCitiesTopMenu = () => {

    return <ul className="locations__list tabs__list">
      {cities.map((it, i) => (
        <WrappedCity onCardClick={onCardClick} onCardMouseOut={onCardMouseOut} key={`city-${i}`} cityObject={it} idx={i} />
      )
      )}
    </ul>;
  };

  return <>
  {_renderCitiesTopMenu()}
  </>;
};

CitiesTopMenu.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.shape({
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }).isRequired,
    city: PropTypes.string.isRequired,
  })).isRequired,
  onCardMouseOut: PropTypes.func,
  onCardClick: PropTypes.func,
};

export {CitiesTopMenu};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  cities: getCities(state),
});

export default connect(
    mapStateToProps
)(CitiesTopMenu);
