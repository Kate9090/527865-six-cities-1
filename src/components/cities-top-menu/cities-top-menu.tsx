import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';

import City from '../city/city.jsx';
import {getCities} from '../../reducer/data/selectors';


const CitiesTopMenu = (props) => {
  const {cities, onCardClick, onCardMouseOut, current} = props;

  const _renderCitiesTopMenu = () => {

    return <ul className="locations__list tabs__list">
      {cities.map((it, i) => (
        <City cities={cities} onCardClick={onCardClick} current={current} onCardMouseOut={onCardMouseOut} key={`city-${i}`} city={it} idx={i} />
      )
      )}
    </ul>;
  };

  return <>
  {_renderCitiesTopMenu()}
  </>;
};

CitiesTopMenu.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string),
  onCardMouseOut: PropTypes.func.isRequired,
  onCardClick: PropTypes.func.isRequired,
  current: PropTypes.string,
};

export {CitiesTopMenu};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  cities: getCities(state),
});

export default connect(
    mapStateToProps
)(CitiesTopMenu);
