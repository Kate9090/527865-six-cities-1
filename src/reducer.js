
const initialState = {
  city: `Amsterdam`,
  offerInCity: require(`./mocks/offers-in-amsterdam`).offerHotelList,
};


const ActionCreator = ({
  'NEW_CITY': (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),

  // 'RESET': () => ({
  //   type: `RESET`,
  // }),

});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return Object.assign({}, state, {
        city: action.payload,
        offerInCity: require(`./mocks/offers-in-${state.city}`).offer,
      });
    // case `RESET`:
    //   return Object.assign({}, state, initialState);
  }

  return state;
};

export {reducer, ActionCreator};
