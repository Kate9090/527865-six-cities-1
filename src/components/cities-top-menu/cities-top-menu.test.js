import React from "react";
import renderer from 'react-test-renderer';

import {CitiesTopMenu} from './cities-top-menu.jsx';

import {Provider} from 'react-redux';
import {reducer} from '../../reducer';
import {createStore} from 'redux';

const store = createStore(
    reducer
);

import offerCities from '../../mocks/offers-city';
const mock = offerCities;

it(`renders correctly City List`, () => {
  const tree = renderer
    .create(<Provider store={store}><CitiesTopMenu
      cities = {mock}
    /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
