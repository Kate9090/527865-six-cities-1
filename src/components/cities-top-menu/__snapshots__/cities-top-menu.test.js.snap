// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly City List 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        London
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Moscow
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Tokio
      </span>
    </a>
  </li>
</ul>
`;
